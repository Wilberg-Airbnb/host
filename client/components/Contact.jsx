import React from 'react';
import { Item} from 'semantic-ui-react';
import Languages from './Languages.jsx';
import SecurityRoundedIcon from '@material-ui/icons/SecurityRounded';

const contactButtonStyle = {
  padding: '13px 23px',
  fontColor: 'black',
  borderRadius: '8px',
  borderWidth: '1px',
  cursor: 'pointer',
  background: 'transparent'
}

const Contact = ({languages, responseRate, responseTime}) => (
  <Item.Group style={{padding: '0px'}}>
    {languages.length > 0 ? <Languages languages={languages}/> : null}
    <Item>
      <Item.Content>
        Response rate: {responseRate}%
      </Item.Content>
    </Item>
    <Item>
      <Item.Content>
        Response time: {responseTime}
      </Item.Content>
    </Item>
    <Item style={{marginTop: '32px'}}>
      <Item.Content>
      <button style={contactButtonStyle}><h3>Contact host</h3></button>
      </Item.Content>
    </Item>
    <Item style={{marginTop: '24px'}}>
      <Item.Content style={{display: 'flex'}}>
      <SecurityRoundedIcon style={{marginRight: '16px', display: 'block'}}/>
      To Protect your payment, never transfer money or communicate outside of the Airbnb website or app.
      </Item.Content>
    </Item>
  </Item.Group>

)

export default Contact;