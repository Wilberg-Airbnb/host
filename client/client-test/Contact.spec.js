import React from 'react';
import Contact from '../components/Contact';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
configure({ adapter: new Adapter() });
 
describe('Renders Components', () => {
  let component;
  let props = {
    langauges: [],
    responseRate: 88,
    responseTime: 'within an hour'
  }
  beforeEach( () => {
    component = shallow(<Contact languages={props.langauges} responseRate={props.responseRate} duringYourStay={props.duringYourStay} responseTime={props.responseTime}/>)
  });

  test("should render initial layout", () => {
    const component = shallow(<Contact languages={props.langauges} responseRate={props.responseRate} duringYourStay={props.duringYourStay} responseTime={props.responseTime} />);
    expect(component.getElements()).toMatchSnapshot();
  });

});


