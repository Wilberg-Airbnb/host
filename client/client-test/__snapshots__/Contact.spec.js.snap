// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Components should render initial layout 1`] = `
Array [
  <ItemGroup>
    <Item>
      <ItemContent>
        Response rate: 
        88
        %
      </ItemContent>
    </Item>
    <Item>
      <ItemContent>
        Response time: 
        within an hour
      </ItemContent>
    </Item>
    <Item
      style={
        Object {
          "marginTop": "32px",
        }
      }
    >
      <ItemContent>
        <button
          style={
            Object {
              "background": "transparent",
              "borderRadius": "8px",
              "borderWidth": "1px",
              "cursor": "pointer",
              "fontColor": "black",
              "padding": "13px 23px",
            }
          }
        >
          <h3>
            Contact host
          </h3>
        </button>
      </ItemContent>
    </Item>
    <Item
      style={
        Object {
          "marginTop": "24px",
        }
      }
    >
      <ItemContent
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <Memo(SecurityRoundedIcon)
          style={
            Object {
              "display": "block",
              "marginRight": "16px",
            }
          }
        />
        To Protect your payment, never transfer money or communicate outside of the Airbnb website or app.
      </ItemContent>
    </Item>
  </ItemGroup>,
]
`;
